plugins/modules/inventory_workflow_manager.py

```diff
@@ -2587,18 +2785,36 @@ def get_provision_wired_device(self, device_ip):
         """
 
         try:
-            flag = 3
-            response = self.dnac._exec(
-                family="sda",
-                function='get_provisioned_wired_device',
-                op_modifies=True,
-                params={"device_management_ip_address": device_ip}
-            )
-
-            if response.get("status") == "success" and "retrieved successfully" in response.get("description"):
-                flag = 2
-                self.log("Wired device '{0}' already provisioned in the Cisco Catalyst Center.".format(device_ip), "INFO")
-
+            if self.dnac_version <= self.version_2_3_5_3:
+                flag = 3
+                response = self.dnac._exec(
+                    family="sda",
+                    function='get_provisioned_wired_device',
+                    op_modifies=True,
+                    params={"device_management_ip_address": device_ip}
+                )
+                self.log("Received API response from 'get_provisioned_wired_devices': {0}".format(str(response)), "DEBUG")
+                if response.get("status") == "success" and "retrieved successfully" in response.get("description"):
+                    flag = 2
+                    self.log("Wired device '{0}' already provisioned in the Cisco Catalyst Center.".format(device_ip), "INFO")
+
+            if self.dnac_version >= self.version_2_3_7_6:
+                flag = 3
+                device_ids = self.get_device_ids([device_ip])
+                device_id = device_ids[0]
+                self.log("device_id")
```

Comment:
self.log("Device ID for IP {0}: {1}".format(device_ip, device_id), "DEBUG")

Reply:
updated the code